### DISCUSSION TITLE
Could CommandPalette post "highlighted" events to the App when you move the cursor through the result list?

### DISCUSSION THREAD

**Discussion by darrenburns** [Textualize/textual#4587] [Ideas]
A fix was put in place a while back to prevent `OptionList.Highlighted` events from "escaping" the command palette.

I'm interested in receiving these events inside my app - I wonder if the command palette could post it's own `CommandPalette.ResultHighlighted` event.

The reason I want this is when an option is highlighted, I want to update my app to preview what it looks like. In VS Code, when you "highlight" a theme in the command palette, the full UI updates to give you a preview of how it'll look - I'd like a similar feature in my app.

If there are no objections - I can PR this.

**Comment by davep** [comment]
It's a bit niche but it makes sense to me: information isolation over information loss.

🤔 . o O ( I wonder if it could/should also be an `App` signal that a screen can subscribe to )

**Comment by willmcgugan** [comment]
Seems reasonable. Although where would that message go? To the app?

Would you also need a message for when the command palette is dismissed?

**Reply by darrenburns** [reply]
I was thinking to the app yeah

**Reply by darrenburns** [reply]
Yeah I'd need one for dismissing it too
